src/Compilers/Core/Portable/Operations/OperationInterfaces.xml

```diff
@@ -2036,6 +2035,13 @@
         </summary>
       </Comments>
     </Property>
+    <Property Name="UsingKind" Type="UsingKind">
```

Comment:
I don't think this is what I'd do either. `UsingKind` limits future extensibility in the same was that the two flag bools do. Rather, I'd suggest something like `DeclarationKind`, with options of `Local`, `Using`, and `AsyncUsing`.

Reply:
For example, there's an open request for a `fixed` declaration statement, that would fit in nicely with this API.

Reply:
Any reason these aren't just bools?  `IsUsing` `IsAsync`?  (and, in the future `IsFixed`)?

Reply:
`DeclarationKind` is problematic because it already exists as public in Microsoft.CodeAnalysis.Editing  
`LocalDeclarationKind` already exists in Microsoft.CodeAnalysis.VisualBasic.Symbols although thats `Friend` so we could conceivable promote it and re-use it for this purpose?

We could do the two bools solution, but is a bit odd as you can't have IsAsync be true without IsUsing.

Reply:
@333fred i only know DeclarationKind already exists because that's how I first defined it. Non clashing name suggestions welcome, and i'll happily update

Reply:
@CyrusNajmabadi https://github.com/dotnet/roslyn/pull/39125#discussion_r333158890.
I'll mull on the DeclarationKind naming.

Reply:
Actually, I thought it collided a lot more than it did. I've changed it to `DeclarationKind`

Reply:
I don't feel hugely strongly.  I'm personally ok with booleans where one implies the other.  But i'm not going to die on this hill :)